Support click house as data source #202

dougkamp opened the pull request:
## Pull request type

<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. -->

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ X] Feature
- [ ] Other (please describe):

## Issue

Resolves #202 

## Description of change & new behavior

Now adds support for ClickHouse data warehouse (needs further work - see below)

## Screenshots

_For frontend updates, please include a screenshot._

## Technical Spec/Implementation Notes

Implements ClickHouse DW support using clickhouse-sqlalchemy package. 

Note: I have been unable to get a successful connection on my local dev PC running ClickHouse in Docker as a container. I have spent many hours trying to work out why and at this stage I am stuck. I am not sure if its a local dev environment issue, a driver (clickhouse-sqlalchemy) issue, a bug or some combination of some or all of these. It needs someone more familiar with ClickHouse and SqlAlchemy to nut this one out unfortunately

Please check if your PR fulfills the following requirements:

- [ ] If it's a backend change, tests for the changes have been added and run successfully.
- [ ] If it's a frontend change, Prettier has been run & all tests pass

dougkamp committed:
Merge pull request #2 from monosidev/develop

Develop

dougkamp committed:
Initial addition of clickhouse, more work to do

dougkamp committed:
Further functionality and bug fixes

dougkamp committed:
minor update

Slach commented:
@dougkamp i can help with docker + clickhouse
could you provide more context and share error messages?
i don't see any changes in docker-compose related files which will try to clickhouse service

dougkamp commented:
> 

Hi @Slach Thanks for the offer but i have just managed to get it working. It was a Docker networking config error in my ClickHouse container when i set it up (i wasn't correctly mapping the internal port to the external port i needed to connect). 

Now i just need to fix the SQL errors I am getting for all the measure and should be good to go.

dougkamp committed:
Fixes to SQL queries

iporollo reviewed (approved):
this is awesome, looks good to me

dougkamp committed:
Final commit after fixing some bugs in SQL syntax

dougkamp commented:
Some fixes to Clickhouse support addressing issues found during testing of metrics queries as Clcikhouse internal table structure and SQL query syntax is quite different to MS SQL, PostgreSQL etc.
